spec/components/email/receiver_spec.rb

```diff
@@ -985,6 +985,93 @@ def create_post_reply_key(value)
       end
     end
 
+    context "emailing a group by email_username and following reply flow" do
+      let!(:topic) do
+        group.update(email_username: "team@somesmtpaddress.com")
+        process(:email_to_group_email_username_1)
+        Topic.last
+      end
+      let(:group_user) do
+        g_user = Fabricate(:user)
+        Fabricate(:group_user, user: g_user, group: group)
+        g_user
+      end
+
+      before do
+        NotificationEmailer.enable
+        Jobs.run_immediately!
+      end
+
+      def reply_as_group_user
+        group_post = PostCreator.create(
+          group_user,
+          raw: "Thanks for your request. Please try to restart.",
+          topic_id: topic.id
+        )
+        email_log = EmailLog.last
+        [email_log, group_post]
+      end
+
+      it "creates an incoming email and topic correctly, and adds the group to the topic" do
+        incoming = IncomingEmail.find_by(topic: topic)
+        user = User.find_by_email("two@foo.com")
+        expect(topic.topic_allowed_users.first.user_id).to eq(user.id)
+        expect(topic.topic_allowed_groups.first.group_id).to eq(group.id)
+        expect(incoming.to_addresses).to eq("team@somesmtpaddress.com")
+        expect(incoming.from_address).to eq("two@foo.com")
+        expect(incoming.message_id).to eq("u4w8c9r4y984yh98r3h69873@foo.bar.mail")
+      end
+
+      it "creates an EmailLog when someone from the group replies, and does not create an IncomingEmail record for the reply" do
+        email_log, group_post = reply_as_group_user
+        expect(email_log.message_id).to eq("topic/#{topic.id}/#{group_post.id}@test.localhost")
+        expect(email_log.to_address).to eq("two@foo.com")
+        expect(email_log.email_type).to eq("user_private_message")
+        expect(email_log.post_id).to eq(group_post.id)
+        expect(IncomingEmail.find_by(post_id: group_post.id)).to eq(nil)
+      end
+
+      it "processes a reply from the OP user to the group SMTP username, linking the reply_to_post_number correctly by
+      matching in_reply_to to the email log" do
+        email_log, group_post = reply_as_group_user
+
+        reply_email = email(:email_to_group_email_username_2)
+        reply_email.gsub!("MESSAGE_ID_REPLY_TO", email_log.message_id)
+        Email::Receiver.new(reply_email).process!
+
+        reply_post = Post.last
+        expect(reply_post.reply_to_user).to eq(group_user)
+        expect(reply_post.reply_to_post_number).to eq(group_post.post_number)
```

Comment:
This is the relevant test to check the reply_to_user and reply_to_post_number is correct. I added a whole flow test to make sure it was all working correctly, in my head and in the code, rather than just testing for this one specific case.